packages/location-field/src/index.tsx

```diff
@@ -179,72 +201,143 @@ const LocationField = ({
+                geocodedFeatures = [];
+              } else {
+                const {
+                  otherFeatures,
+                  stationFeatures,
+                  stopFeatures
+                } = getFeaturesByCategoryWithLimit(
+                  geocodedFeatures,
+                  suggestionCount,
+                  sortByDistance,
+                  preferredLayers
+                );
+                // Breakdown results found by type.
+                const parts = [];
+                if (stopFeatures.length) {
+                  parts.push(
+                    intl.formatMessage(
+                      {
+                        description: "Shows the count of transit stops",
+                        id: "otpUi.LocationField.stopCount"
+                      },
+                      { count: stopFeatures.length }
+                    )
+                  );
+                }
+                if (stationFeatures.length) {
+                  parts.push(
+                    intl.formatMessage(
+                      {
+                        description: "Shows the count of stations",
+                        id: "otpUi.LocationField.stationCount"
+                      },
+                      { count: stationFeatures.length }
+                    )
+                  );
+                }
+                if (otherFeatures.length) {
+                  parts.push(
+                    intl.formatMessage(
+                      {
+                        description: "Shows the count of other places",
+                        id: "otpUi.LocationField.otherCount"
+                      },
+                      { count: otherFeatures.length }
+                    )
+                  );
+                }
+                const hasResults = parts.length !== 0;
+                const results = hasResults
+                  ? intl.formatList(parts, { type: "conjunction" })
+                  : intl.formatMessage({
+                      description: "Indicates no results",
+                      id: "otpUi.LocationField.noResults"
+                    });
+                const resultsFoundText = intl.formatMessage(
+                  {
+                    description: "Text about geocoder results found",
+                    id: "otpUi.LocationField.resultsFound"
+                  },
+                  {
+                    input: text,
+                    results
+                  }
+                );
+                if (hasResults) {
+                  // If there are results, concatenate sentences about results found and
+                  // instructions for assistive technology users on how to access results.
+                  const instructions = intl.formatMessage({
+                    description: "Instructions on accessing geocoder results",
+                    id: "otpUi.LocationField.howToAccessResults"
+                  });
+                  message = `${resultsFoundText} ${instructions}`;
```

Comment:
Another one -- is this i18n compatible?

Reply:
This one should work because it joins two independent sentences.